settings_backup.php

```diff
@@ -1,37 +1,234 @@
+        die('Failed to create backups folder inside uploads');
+    }
+}
+
+$backups = array_diff(scandir($backupFolder), array('..', '.'));
+
+// Database connection
+$mysqli = mysqli_connect($dbhost, $dbusername, $dbpassword, $database) or die('Database Connection Failed');
+$conn = new mysqli($dbhost, $dbusername, $dbpassword, $database);
+
+// Handle backup action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['backup'])) {
+    // Create a backup
+    $backupFileName = date("d-m-Y_H-i-s") . ".sql";
+    $backupPath = $backupFolder . $backupFileName;
+
+    // Run mysqldump command to include table content
+    $escapedBackupPath = escapeshellarg($backupPath);
+    $command = "mysqldump --complete-insert --skip-comments -h $dbhost -u $dbusername -p$dbpassword $database > $escapedBackupPath";
+    exec($command);
+
+    // Refresh backup list after creating a new backup
+    $backups = array_diff(scandir($backupFolder), array('..', '.'));
+}
+
+// Handle restore action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['proceed-restore'])) {
+    $selectedBackup = $_POST['proceed-restore'];
+
+    // Use realpath to get the canonicalized absolute pathname
+    $sqlFile = realpath($backupFolder . $selectedBackup);
+
+    // Check if the obtained path is within the allowed directory
+    if ($sqlFile !== false && strpos($sqlFile, realpath($backupFolder)) === 0) {
+        $sqlContent = file_get_contents($sqlFile);
+
+        // Remove comments and split into separate queries
+        $sqlQueries = preg_split('/;(?=(?:[^\'"]*[\'"][^\'"]*[\'"])*[^\'"]*$)/', $sqlContent);
+
+        foreach ($sqlQueries as $query) {
+            $query = trim($query);
+            if (!empty($query)) {
+                // Execute each query separately using $conn
+                $result = $conn->query($query);
+
+                // Check for execution success
+                if ($result === false) {
+                    die("Error executing query: " . $conn->error);
+                }
+            }
+        }
+
+        // Display success message
+        echo '<div class="alert alert-success" role="alert">Database restore successful!</div>';
+    } else {
+        // Log an error or take appropriate action for invalid paths
+        echo 'Invalid backup path: ' . $sqlFile;
+    }
+}
+
+// Handle delete action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['delete'])) {
+    $selectedBackup = $_POST['delete'];
+
+    // Validate the selectedBackup variable to prevent directory traversal
+    if (in_array($selectedBackup, $backups)) {
+        unlink($backupFolder . $selectedBackup);
+    }
+}
+
+// Handle delete selected action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['delete-selected'])) {
+    // Implement delete selected logic here
+    if (isset($_POST['selectedBackups'])) {
+        foreach ($_POST['selectedBackups'] as $selectedBackup) {
+            unlink($backupFolder . $selectedBackup);
```

Comment:
changed with:
              // Handle delete selected action
              if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['delete-selected'])) {
             // Implement delete selected logic here
             if (isset($_POST['selectedBackups'])) {
             foreach ($_POST['selectedBackups'] as $selectedBackup) {
            $backupPath = $backupFolder . $selectedBackup;

            // Validate the file path to prevent directory traversal
            if (is_file($backupPath) && strpos(realpath($backupPath), realpath($backupFolder)) === 0) {
                unlink($backupPath);
            } else {
                // Log an error or take appropriate action for invalid paths
                echo 'Invalid backup path: ' . $backupPath;
            }
        }
    }
}